src/Monetization.php

```diff
@@ -196,6 +198,72 @@ public function getCompanyPrepaidBalances(CompanyInterface $company, \DateTimeIm
     return $this->getPrepaidBalances($balance_controller, $billingDate);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function isLatestTermsAndConditionAccepted(string $developer_id): ?bool {
+    try {
+      $latest_tnc_id = $this->getLatestTermsAndConditions()->id();
+      if ($history = $this->sdk_controller_factory->developerTermsAndConditionsController($developer_id)->getTermsAndConditionsHistory()) {
+        foreach ($history as $item) {
+          $tnc = $item->getTnc();
+          if ($tnc->id() === $latest_tnc_id) {
+            return ($item->getAction() === 'ACCEPTED');
+          }
+        }
+      }
+    }
+    catch (\Throwable $t) {
+      // @TODO: Log errors ?
+    }
+
+    return NULL;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLatestTermsAndConditions(): ?TermsAndConditions {
+    if ($terms = $this->sdk_controller_factory->termsAndConditionsController()->getPaginatedEntityList()) {
+      $inEffectTerms = [];
+      // Loop through array of terms to pull only terms that are in effect.
+      foreach ($terms as $tnc) {
+        $effectiveDate = $tnc->getStartDate();
+        if ($effectiveDate < new \DateTimeImmutable('now', $effectiveDate->getTimezone())) {
+          array_push($inEffectTerms, $tnc);
+        }
+      }
+      // Make sure to sort terms and conditions by date.
+      usort($inEffectTerms, function ($a, $b) {
+        // It is easier to compare unix timestamps.
+        return $a->getStartDate()->format('U') - $b->getStartDate()->format('U');
+      });
+      $ids = array_keys($inEffectTerms);
+      return $inEffectTerms[end($ids)] ?? NULL;
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function acceptLatestTermsAndConditions(string $developer_id): ?LegalEntityTermsAndConditionsHistoryItem {
+    try {
+      return $this->sdk_controller_factory->developerTermsAndConditionsController($developer_id)
+        ->acceptTermsAndConditionsById($this->getLatestTermsAndConditions()->id());
+    }
+    catch (\Throwable $t) {
+      // @TODO: Log errors ?
```

Comment:
We shouldn't just swallow exceptions, need ticket to work on this later

Reply:
I created an issue.